controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -246,17 +259,51 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 			// if we found more than one CephBlockPool, we can't determine which one to select, so error out
 			cbpItemsLen := len(cephBlockPoolList.Items)
 			if cbpItemsLen == 0 {
-				cbpNewName := fmt.Sprintf("cephblockpool-%s-%s", r.storageConsumer.Name, generateUUID())
+				cbpNewName := fmt.Sprintf("cephblockpool-profile-%s", storageProfile.Name)
 				r.log.V(1).Info("no valid CephBlockPool found, creating new one", "CephBlockPool", cbpNewName)
 				r.cephBlockPool.Name = cbpNewName
 			} else if cbpItemsLen == 1 {
 				r.cephBlockPool.Name = cephBlockPoolList.Items[0].GetName()
 				r.log.V(1).Info("valid CephBlockPool found", "CephBlockPool", r.cephBlockPool.Name)
 			} else {
-				return fmt.Errorf("invalid number of CephBlockPools for storage consumer %q and storage profile %q: found %d, expecting 0 or 1", r.storageConsumer.Name, storageProfile.Name, cbpItemsLen)
+				return fmt.Errorf("invalid number of CephBlockPools for storage profile %q: found %d, expecting 0 or 1", storageProfile.Name, cbpItemsLen)
+			}
+		}
+
+		// check if a CephBlockPoolRadosNamespace resource exists for the desired storageconsumer and storageprofile.
+		if r.cephRadosNamespace.Name == "" {
+			cephRadosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+			err := r.Client.List(
+				r.ctx,
+				cephRadosNamespaceList,
+				client.InNamespace(r.OperatorNamespace),
+				client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)})
+			if err != nil {
+				return err
+			}
+
+			// if we found no CephBlockPoolRadosNamespaces, generate a new name
+			// if we found only one CephBlockPoolRadosNamespace with our query, we're good
+			// if we found more than one CephBlockPoolRadosNamespace, we can't determine which one to select, so error out
+			rnsItemsLen := len(cephRadosNamespaceList.Items)
+			if rnsItemsLen == 0 {
+				rnsNewName := fmt.Sprintf("cephradosnamespace-%s-%s", r.cephBlockPool.Name, r.storageConsumer.Name)
```

Comment:
With this approach you can hit the name length limit. To solve this we usually use a simple hash for the different parts. So the resulting name will be `cephradosnamespace-<has from blockpool name and storage consumer name>`